diff --git a/src/parser/parse_generic.c b/src/parser/parse_generic.c
index feeedc9..3214e57 100644
--- a/src/parser/parse_generic.c
+++ b/src/parser/parse_generic.c
@@ -15,7 +15,7 @@ static inline void please_parse_array_field(cJSON *json_aray, struct s_field_inf
 				printf("FUCKING ARR TYPE%d\n", field_info->arr_type);
 				exit(-6);
 			}
-		pos[pos_idx] = ((float)tmp->valuedouble);
+		pos[pos_idx] = ((float)tmp->valuefloat);
 		// if (pos[pos_idx] < 0 && !(field_info->can_be_signed))
 		// {
 		// 	printf("THIS FIELD CAN'T BE SIGNED!\n");
@@ -55,9 +55,9 @@ void please_parse_field(cJSON *json_chain, struct s_field_info *field_info, void
 		// }
 		field_info->array_size = 1;
 		if (field_info->is_int)
-			please_validate_and_save((void*)&json_field->valueint, _addr, field_info);
+			please_validate_and_save((void*)(&json_field->valueint), _addr, field_info);
 		else
-			please_validate_and_save((void*)&json_field->valuedouble, _addr, field_info);
-        printf("field: %f\n", json_field->valuedouble);
+			please_validate_and_save((void*)(&json_field->valuefloat), _addr, field_info);
+        printf("field: %f\n", json_field->valuefloat);
 	}
 }
\ No newline at end of file
